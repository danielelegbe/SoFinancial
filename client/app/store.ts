import { configureStore } from '@reduxjs/toolkit';
import userReducer from '../features/user/userSlice';
import chatReducer from '../features/chat/chatSlice';

export const store = configureStore({
  reducer: {
    user: userReducer,
    chat: chatReducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
