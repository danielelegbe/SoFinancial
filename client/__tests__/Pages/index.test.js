import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import  Home from '../../pages/index';
import { store } from '../../app/store';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { afterEach, describe } from 'jest-circus';

describe('Home page tests', () => {
  let mockArticle
  beforeEach(() => {
    mockArticle = {
      author: 'An author',
      rights: 'No rights',
      link: 'No links',
      media: 'No media',
      title: 'No titles',
      _id: 'No ID',
    }
  })
  afterEach(() => {
    cleanup()
  })

  test('It should render the index page', () => {
    render(<Provider store = {store}><Home uniqueArticles = {mockArticle}/></Provider>)
      const home = screen.getByTestId('home');
      expect(home).toBeInTheDocument();
  })

  test('snapshot home page test', () => {
    const homePage = renderer.create(<Provider store = {store}><Home uniqueArticles = {mockArticle}/></Provider>).toJSON();
    expect(homePage).toMatchSnapshot()
  })
})