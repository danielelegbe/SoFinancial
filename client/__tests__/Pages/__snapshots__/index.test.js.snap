// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot home page test 1`] = `
Array [
  <h2
    className="chakra-heading css-0"
    data-testid="home"
  >
    Home
  </h2>,
  <div
    className="css-j7qwjs"
  >
    <div
      className="css-kdwx3d"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        className="css-1demtxk"
        data-indeterminate=""
        role="progressbar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
  </div>,
]
`;
